reposilite-backend/src/main/kotlin/com/reposilite/maven/Repository.kt

```diff
@@ -56,11 +57,12 @@ class Repository internal constructor(
         val sha1 = location.resolveSibling(location.getSimpleName() + ".sha1")
         val sha256 = location.resolveSibling(location.getSimpleName() + ".sha256")
         val sha512 = location.resolveSibling(location.getSimpleName() + ".sha512")
+        val byteArray = IOUtils.toByteArray(bytes)
 
-        return storageProvider.putFile(md5, DigestUtils.md5Hex(bytes).byteInputStream())
-            .flatMap { storageProvider.putFile(sha1, DigestUtils.sha1Hex(bytes).byteInputStream()) }
-            .flatMap { storageProvider.putFile(sha256, DigestUtils.sha256Hex(bytes).byteInputStream()) }
-            .flatMap { storageProvider.putFile(sha512, DigestUtils.sha512Hex(bytes).byteInputStream()) }
+        return storageProvider.putFile(md5, DigestUtils.md5Hex(byteArray).byteInputStream())
+            .flatMap { storageProvider.putFile(sha1, DigestUtils.sha1Hex(byteArray).byteInputStream()) }
+            .flatMap { storageProvider.putFile(sha256, DigestUtils.sha256Hex(byteArray).byteInputStream()) }
+            .flatMap { storageProvider.putFile(sha512, DigestUtils.sha512Hex(byteArray).byteInputStream()) }
     }
```

Comment:
We can't store the whole file in memory, so I'd put the input bytes in the temporary file first. Then, we could just pass a new InputStream to the temp file in each of those method calls. Thanks to that, we can also avoid bringing up additional dependency :)

Reply:
Good call. Should be fixed now.